### PR TITLE
feat: add `@sequelize/utils` package

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description of Changes

This PR adds a new package, `@sequelize/utils`, which is meant to expose the utility functions we use in more than one package.

It exposes code that is not Sequelize specific. This package should be usable in any project.

### CODE REVIEW

**Review by WikiRik — APPROVED** [review]
Haven't taken an in depth look at some of the utils, but it seems like there's plenty of tests added (even though they're structured differently than the other packages). Changes to core and validator look good

**Comment by WikiRik** [comment]
Do we need to change any files to allow the new package to be published? Either way; I'll let you merge this so you can check if releasing the package is going well

**Comment by ephys** [comment]
> Do we need to change any files to allow the new package to be published? 

I think we're good but I'll do a test release and we'll find out

**Review by WikiRik — APPROVED** [review]

**Pull request merged**
